lib/plug/conn.ex

```diff
@@ -460,12 +460,15 @@ defmodule Plug.Conn do
   otherwise `{:error, reason}`.
   """
   @spec chunk(t, body) :: {:ok, t} | {:error, term} | no_return
-  def chunk(%Conn{state: :chunked} = conn, ""), do: {:ok, conn}
   def chunk(%Conn{adapter: {adapter, payload}, state: :chunked} = conn, chunk) do
-    case adapter.chunk(payload, chunk) do
-      :ok                  -> {:ok, conn}
-      {:ok, body, payload} -> {:ok, %{conn | resp_body: body, adapter: {adapter, payload}}}
-      {:error, _} = error  -> error
+    case IO.iodata_length(chunk) do
```

Comment:
Thanks for doing that, @jeregrine!  I'm curious what kind of `iodata` you used in your benchmark?  For small, simple examples, I'd absolutely expect `IO.iodata_length` to be optimal given it's written in C.  My concern is for large chunks.  For example, consider an app that is streaming a CSV as it is generated , sending chunks of 100K rows at a time, where it is emitting iodata chunks as nested lists (one list entry per row where each row is a list containing one entry per column).  For such a case, `iodata_length` is going to have to traverse the nested lists to build the binary.  Plus it's generating a lot of garbage if it has to generate the entire binary just to get the length.

An `empty_iodata?` function--whether pre-existing or hand-written--that short-circuited as soon as it found a non-empty binary at some level of nesting in the `iodata`, would be noticeably faster, I expect.